services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationController.kt

```diff
@@ -65,6 +68,16 @@ class ApplicationController(
     fun getById(@PathVariable(name = "id") id: Long): Application {
         val userId = currentUserId()
         val application = service.getApplicationById(id, userId)
+        val hankeTunnus = application.hankeTunnus ?: throw HankeNotFoundException("")
+        val hankeId = hankeService.getHankeId(hankeTunnus)
+
+        if (
+            hankeId == null ||
+                !permissionService.hasPermission(hankeId, userId, PermissionCode.EDIT_APPLICATIONS)
```

Comment:
I think the permission for viewing an application should be just `VIEW`. There's no permission specifically for viewing an applications, though there could be. The current specs say that everyone with any kind of permissions has also the permission to view applications. (https://helsinkisolutionoffice.atlassian.net/wiki/spaces/HAI/pages/8212906098/K+ytt+oikeustasojen+toimintojen+m+rittely)

Reply:
Thanks for clarification. Changed.